kgforge/specializations/models/rdf/directory_service.py

```diff
@@ -61,30 +66,47 @@ def resolve_context(self, iri: str) -> Dict:
     def generate_context(self) -> Dict:
         return self._generate_context()
 
-    def _build_shapes_map(self) -> Dict:
-        query = """
-            PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
-            PREFIX sh: <http://www.w3.org/ns/shacl#>
-            SELECT ?type ?shape WHERE {
-                { ?shape sh:targetClass ?type .}
-                UNION {
-                    SELECT (?shape as ?type) ?shape WHERE {
-                        ?shape a sh:NodeShape .
-                        ?shape a rdfs:Class
-                    }
-                }
-            } ORDER BY ?type"""
+    def _build_shapes_map(self) -> Tuple[Dict, Dict, Dict]:
+        query = build_shacl_query(
+            defining_property_uri=self.NXV.shapes,
+            deprecated_property_uri=OWL.deprecated,
+            optional=True,
+            search_in_graph=True,
+            context=self.context,
+        )
         res = self._graph.query(query)
-        return {row["type"]: row["shape"] for row in res}
+        class_to_shape = {}
+        shape_to_defining_resource = {}
+        shape_to_named_graph = {}
+        defining_resource_to_named_graph = {}
+        for row in res:
+            class_to_shape[URIRef(row["type"])] = URIRef(row["shape"])
+            shape_to_defining_resource[URIRef(row["shape"])] = URIRef(
+                row["resource_id"]
+            )
+            shape_to_named_graph[URIRef(row["shape"])] = URIRef(row["g"])
+            defining_resource_to_named_graph[URIRef(row["resource_id"])] = URIRef(
+                row["g"]
+            )
+        return (
+            class_to_shape,
+            shape_to_defining_resource,
+            shape_to_named_graph,
+            defining_resource_to_named_graph,
+        )
+
+    def load_shape_graph(self, graph_id, schema_id):
+        return self._graph.graph(rdflib.term.URIRef(graph_id))
```

Comment:
change `rdflib.term.URIRef` to `URIRef` as it is already imported that way, to be consistent with previous lines

Reply:
Done

Reply:
Both done